src/test/java/edu/harvard/iq/dataverse/api/DatasetsIT.java

```diff
@@ -718,11 +703,83 @@ public void testDatasetVersionsAPI() {
+            then().assertThat().statusCode(OK.getStatusCode()).body("data[0].files", equalTo(null));
+        UtilIT.getDatasetVersion(datasetPid, DS_VERSION_LATEST_PUBLISHED, apiTokenNoPerms, excludeFiles, includeDeaccessioned).
+            then().assertThat().statusCode(OK.getStatusCode()).body("data[0].files", equalTo(null));
+
+        excludeFiles = false;
+        UtilIT.getDatasetVersion(datasetPid, DS_VERSION_LATEST_PUBLISHED, apiToken, excludeFiles, includeDeaccessioned).
+            then().assertThat().statusCode(OK.getStatusCode()).body("data.files.size()", equalTo(1));
+        UtilIT.getDatasetVersion(datasetPid, DS_VERSION_LATEST_PUBLISHED, apiTokenNoPerms, excludeFiles, includeDeaccessioned).
+            then().assertThat().statusCode(OK.getStatusCode()).body("data.files.size()", equalTo(1));
+
+        
+        //We deaccession the dataset
+        Response deaccessionDatasetResponse = UtilIT.deaccessionDataset(datasetId, DS_VERSION_LATEST_PUBLISHED, "Test deaccession reason.", null, apiToken);
+        deaccessionDatasetResponse.then().assertThat().statusCode(OK.getStatusCode());
+
+        //Set of tests on deaccesioned dataset, only 3/9 should return OK message
+
+        includeDeaccessioned = true;
+        excludeFiles = true;
+        UtilIT.getDatasetVersion(datasetPid, DS_VERSION_LATEST_PUBLISHED, apiToken, excludeFiles, includeDeaccessioned).
+            then().assertThat().statusCode(OK.getStatusCode()).body("data[0].files", equalTo(null));
+        UtilIT.getDatasetVersion(datasetPid, DS_VERSION_LATEST_PUBLISHED, apiTokenNoPerms, excludeFiles, includeDeaccessioned).
+            then().assertThat().statusCode(OK.getStatusCode()).body("data[0].files", equalTo(null));
+        excludeFiles = false;
+        UtilIT.getDatasetVersion(datasetPid, DS_VERSION_LATEST_PUBLISHED, apiToken, excludeFiles, includeDeaccessioned).
+            then().assertThat().statusCode(OK.getStatusCode()).body("data.files.size()", equalTo(1));;
+        UtilIT.getDatasetVersion(datasetPid, DS_VERSION_LATEST_PUBLISHED, apiTokenNoPerms, excludeFiles, includeDeaccessioned).
+            then().assertThat().statusCode(404);
+        
+        includeDeaccessioned = false;
+        excludeFiles = true;
+        UtilIT.getDatasetVersion(datasetPid, DS_VERSION_LATEST_PUBLISHED, apiToken, excludeFiles, includeDeaccessioned).
+            then().assertThat().statusCode(404);
+        UtilIT.getDatasetVersion(datasetPid, DS_VERSION_LATEST_PUBLISHED, apiTokenNoPerms, excludeFiles, includeDeaccessioned).
+            then().assertThat().statusCode(404);
+        excludeFiles = false;
+        UtilIT.getDatasetVersion(datasetPid, DS_VERSION_LATEST_PUBLISHED, apiToken, excludeFiles, includeDeaccessioned).
+            then().assertThat().statusCode(404);
+        UtilIT.getDatasetVersion(datasetPid, DS_VERSION_LATEST_PUBLISHED, apiTokenNoPerms, excludeFiles, includeDeaccessioned).
+            then().assertThat().statusCode(404);
+
+      
+
+
```

Comment:
While we're testing deaccessioning, would it be possible to add some tests asserting search behavior? Please see https://github.com/IQSS/dataverse/issues/10164#issuecomment-1854151124

Reply:
I into this but this should address the issue with the API

Reply:
I did a quick dive into the Search API. It seems to be unrelated to this one and if I get the question right, they want to use the search API to include a deaccessioned dataset but I can't see any parameters similar to `includeDeaccessioned`, probably would be a feature request but still not sure.